scripts/dev/upstream_merge/README.md

```diff
@@ -0,0 +1,144 @@
+# HOW TO USE THE MERGE SCRIPT
+
+This script automates the process of merging upstream changes, building images, and testing them on a virtual machine (VM). Follow the steps below to use the script effectively.
+
+---
+
+## **Steps to Run the Script**
+
+### **1. Navigate to the `nilrt` Directory**
+```bash
+cd ~/nilrt
+```
+
+### **2. Run the Script**
+
+#### **To Perform a Merge**
+This will merge upstream changes, build the images, and test them:
+```bash
+python3 scripts/dev/upstream_merge/upstream_merge.py -c scripts/dev/upstream_merge/automation_conf.json
+```
+
+#### **To Skip the Merge and Only Build and Test**
+If you want to skip the merge step and directly proceed with building and testing:
+```bash
+python3 scripts/dev/upstream_merge/upstream_merge.py -c scripts/dev/upstream_merge/automation_conf.json -skip-merge True
+```
+
+---
+
+## **Configuration File: `automation_conf.json`**
+
+The script relies on a configuration file (`automation_conf.json`) to define various parameters. Below is an explanation of the fields in the configuration file:
+
+- **`conf_file_path`**:  
+  Path to the configuration file (default: `repos.conf`). This file must be located in the `/scripts/dev/upstream_merge/` folder.
+
+- **`force_checkout`**:  
+  Enables forceful checkout to the base branch if set to `True`.
+
+- **`forks`**:  
+  A dictionary where the keys are local repository names (as mentioned in `repos.conf`) and the values are the URLs of your forks.
+
+- **`upstream_repo_name`**:  
+  The name of the upstream remote repository.
+
+- **`merge_branch_name`**:  
+  The name of the branch where the merge will be performed.
+
+- **`email_from`**:  
+  The email address from which the merge report will be sent.
+
+- **`email_to`**:  
+  The email address to which the merge report will be sent.
+
+- **`email_log_level`**:  
+  - `0`: Minimal information will be included in the email.  
+  - `1`: Additional detailed information will be included in the email.
+
+- **`log_level`**:  
+  Integer representing the logging level:  
+  - `10`: DEBUG  
+  - `20`: INFO  
+  - `30`: WARNING  
+  - `40`: ERROR  
+  - `50`: CRITICAL  
+
+- **`work_item_id`**:  
+  The work item ID associated with the pull request.
```

Comment:
Mention that this is a required field that needs to be filled by the user. Also, what happens if the user doesn't fill this field in the JSON?

Is the default 0, and does the PR get created with AB#0 in the description? This is also a good enough way to handle missing inputs, but make sure that it's mentioned in the README.